pkg/pubsub/sqs/subscriber.go

```diff
@@ -0,0 +1,99 @@
+package sqs
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/types"
+
+	"github.com/scribd/go-sdk/pkg/pubsub"
+	"github.com/scribd/go-sdk/pkg/pubsub/pool"
+)
+
+type (
+	Subscriber struct {
+		client      *sqs.Client
+		queueURL    string
+		handler     MsgHandler
+		maxMessages int
+		pool        *pool.Pool
+		waitTime    time.Duration
+	}
+
+	SubscriberConfig struct {
+		SQSClient  *sqs.Client
+		MsgHandler MsgHandler
+		SQSConfig  pubsub.SQS
+	}
+
+	MsgHandler func(msg types.Message)
+)
+
+const (
+	defaultNumWorkers = 1
+	maxWaitTime       = 20 * time.Second
+)
+
+func NewSubscriber(c SubscriberConfig) *Subscriber {
+	workers := c.SQSConfig.Subscriber.Workers
+	if workers == 0 {
+		workers = defaultNumWorkers
+	}
+
+	waitTime := c.SQSConfig.Subscriber.WaitTime
+	if waitTime > maxWaitTime {
+		waitTime = maxWaitTime
+	}
+
+	return &Subscriber{
+		client:      c.SQSClient,
+		handler:     c.MsgHandler,
+		maxMessages: c.SQSConfig.Subscriber.MaxMessages,
+		queueURL:    c.SQSConfig.Subscriber.QueueURL,
```

Comment:
Could we have some validation for these configs, so that it's more miss-configuration proof.